framework/include/systems/NonlinearEigenSystem.h

```diff
@@ -84,7 +84,7 @@ class NonlinearEigenSystem : public NonlinearSystemBase
    */
   virtual SNES getSNES() override;
 
-  TransientEigenSystem & sys() { return _transient_sys; }
+  TransientEigenSystem & sys() const { return _transient_sys; }
```

Comment:
Same comment here. Let's not mark this as `const` because it's returning a non-const reference. All the `const` is doing here is marking the *reference* itself as `const` which is redundant because of course you can never change what a reference is a reference to. If this method was returning a `const TransientEigenSystem &`, then marking it as `const` would be semantically correct.

Reply:
I know I tell you to make everything `const` and here I am telling you to unmark `const` 😆

Reply:
Haha. I have to do case by case